web/src/main/webapp/my-app/marketplace/controllers.js

```diff
@@ -58,9 +58,13 @@ define(['angular', 'jquery', 'require'], function(angular, $, require) {
           && portlet.portletName.indexOf('cms') != -1; // static content portlet
       };
 
+      $scope.GuestMode = function(mainService) {
```

Comment:
:question: `GuestMode` is a factory that takes a service and returns a function?